### PR TITLE
Lightbox Opening Sound

### PR DESCRIPTION
**Pull request by ShoboySnr**
Adds a new feature to allow admins to set a sound to play when the popup opens.

### CODE REVIEW

File: src/assets/js/src/mailoptin.js [hunk 1]
```diff
@@ -1046,7 +1075,7 @@ var mailoptin_optin = {
                             self.addSuccessCloseIcon.call(optin_container);
                         }
 
-                        self.displaySuccessContent.call(optin_container);
+                        self.displaySuccessContent.call(optin_container, response.message);
```
Comment (w3guy) [inline src/assets/js/src/mailoptin.js:1078]: response.message is not used anywhere. Are you sure it is needed here?

File: src/assets/js/src/mailoptin.js [hunk 2]
```diff
@@ -698,9 +695,41 @@ var mailoptin_optin = {
...
+
         $(this).trigger('moOptin:show', [optin_config.optin_uuid, optin_config, optin_type]);
     },
 
+    optin_sound_selected: function (optin_config, optin_type) {
+        //if we are in customizer, don't execute this to avoid double sound playing
+        if ($.MailOptin.is_customize_preview === true) return;
+
+        if(optin_type === 'bar' || optin_type === 'lightbox' || optin_type === 'slidein') {
+            let optin_sound_config = optin_config.optin_sound;
```
Comment (w3guy) [inline src/assets/js/src/mailoptin.js:707]: Please replace every instance of let with var. We are still supporting older browser versions.

File: src/Admin/Customizer/OptinForm/CustomizerControls.php [hunk 3]
```diff
@@ -940,6 +940,35 @@ public function configuration_controls()
...
+                        'section'     => $this->customizerClassInstance->configuration_section_id,
+                        'settings'    => $this->option_prefix . '[optin_sound]',
+                        'description' => __('Select the optin sound to play when Optin is displayed', 'mailoptin'),
+                        'choices'     => [
+                            'none'          => __('None', 'mailoptin'),
+                            'beep.wav'          => __('Beep', 'mailoptin'),
+                            'beep-up.wav'       => __('Beep Up', 'mailoptin'),
+                            'beep-down.wav'     => __('Beep Down', 'mailoptin'),
+                            'clong.wav'         => __('Clong', 'mailoptin'),
+                            'chime.aiff'         => __('Chime', 'mailoptin'),
```
Comment (w3guy) [inline src/Admin/Customizer/OptinForm/CustomizerControls.php:955]: Chime sound isn't working.

File: src/assets/js/customizer-controls/contextual-customizer-controls.js [hunk 4]
```diff
@@ -466,6 +466,26 @@
                 });
             });
 
+            //add event handler to detect when opening sound is changed and play the sound to allow admin to preview it
+            $('select[data-customize-setting-link*=optin_sound]').on('change', function (event) {
```
Comment (w3guy) [inline src/assets/js/customizer-controls/contextual-customizer-controls.js:470]: When a custom sound url is specified, try to trigger the sound too.

File: src/Admin/Customizer/OptinForm/CustomizerControls.php (hunk 3 above)
Comment (w3guy) [reply src/Admin/Customizer/OptinForm/CustomizerControls.php:955]: If you can get Chime to work, maybe replace it with "Pop" => https://freesound.org/people/elmasmalo1/sounds/377017/

And if you can get it to work, add pop regardless.

**Review by w3guy — CHANGES_REQUESTED** [review]

File: src/assets/js/customizer-controls/contextual-customizer-controls.js (hunk 4 above)
Comment (w3guy) [reply src/assets/js/customizer-controls/contextual-customizer-controls.js:471]: @ShoboySnr you didnt fix this

**Review by w3guy — CHANGES_REQUESTED** [review]

File: src/assets/js/customizer-controls/contextual-customizer-controls.js (hunk 4 above)
Comment (ShoboySnr) [reply src/assets/js/customizer-controls/contextual-customizer-controls.js:471]: @w3guy i did fix it. When the custom audio url is entered, it get its value with the selector

File: src/assets/js/customizer-controls/contextual-customizer-controls.js [hunk 5]
```diff
@@ -467,6 +467,43 @@
...
+                var value = $(this).val();
+                if(value !== 'none') {
+                    var audio_sound_url = value !== 'custom' ? mailoptin_globals.public_sound + value : $('li[id*=optin_custom_sound]').find('input').val();
+                    const audio = new Audio( audio_sound_url);
```
Comment (w3guy) [inline src/assets/js/customizer-controls/contextual-customizer-controls.js:475]: @ShoboySnr replace all const with var, please.

File: src/assets/js/customizer-controls/contextual-customizer-controls.js [hunk 6]
```diff
@@ -467,6 +467,43 @@
...
+                    });
+                }
+
+                $('li[id*=optin_custom_sound]').toggle(value === 'custom');
+            }).change();
+
+            //add event handler to detect when a custom sound url is entered and then play the sound
+            $('input[data-customize-setting-link*=optin_custom_sound]').on('change', function (event) {
+                var audio_sound_url = $(this).val();
+                if(audio_sound_url.length > 0) {
```
Comment (w3guy) [inline src/assets/js/customizer-controls/contextual-customizer-controls.js:493]: @ShoboySnr there is code duplication here. I am sure it's possible to use one event listener callback for both situations. For example, you can listen to multiple event like so

```
$('select[data-customize-setting-link*=optin_sound],  input[data-customize-setting-link*=optin_custom_sound]').on('change',
```

**Review by w3guy — CHANGES_REQUESTED** [review]
Great job.

Just one last change and we are done.

**Pull request merged**